File: languages/python/bloock/entity/key/totp_access_control_receipt.py
from typing import List

class TotpAccessControlReceipt:
    """
    Represents a receipt for a Time-based One-Time Password (TOTP) access control.
    """
    def __init__(self, secret: str, secret_qr: str, recovery_codes: List[str]) -> None:
        """
        Creates a new TotpAccessControlReceipt with the provided secret, secret QR code, and recovery codes.
        :type recovery_codes: object
        :type secret_qr: object
        :type secret: object
        :rtype: object
        """
        self.secret = secret
        self.secret_qr = secret_qr
        self.recovery_codes = recovery_codes

File: languages/php/Bloock/Entity/Key/TotpAccessControlReceipt.php
<?php

namespace Bloock\Entity\Key;

use Google\Protobuf\Internal\RepeatedField;

/**
 * Represents a receipt for a Time-based One-Time Password (TOTP) access control.
 */
class TotpAccessControlReceipt
{
    public string $secret;
    public string $secretQr;
    public RepeatedField $recoveryCodes;

    /**
     * Creates a new TotpAccessControlReceipt with the provided secret, secret QR code, and recovery codes.
     * @param string $secret
     * @param string $secretQr
     * @param RepeatedField $recoveryCodes
     */
    public function __construct(string $secret, string $secretQr, RepeatedField $recoveryCodes)
    {
        $this->secret = $secret;
        $this->secretQr = $secretQr;
        $this->recoveryCodes = $recoveryCodes;
    }

    /**
     * Gets the secret of the totp access control receipt.
     * @return string
     */
    public function getSecret(): string
    {
        return $this->secret;
    }

    /**
     * Gets the secret qr of the totp access control receipt.
     * @return string
     */
    public function getSecretQr(): string
    {
        return $this->secretQr;
    }

    /**
     * Gets the recovery codes of the totp access control receipt.
     * @return RepeatedField
     */
    public function getRecoveryCodes(): RepeatedField
    {
        return $this->recoveryCodes;
    }
}
